models/DataObject/Listing/Dao.php

```diff
@@ -78,7 +78,11 @@ public function getTotalCount(): int
         $queryBuilder = $this->getQueryBuilder();
         $this->prepareQueryBuilderForTotalCount($queryBuilder, $this->getTableName() . '.id');
 
-        $totalCount = $this->db->fetchOne((string) $queryBuilder, $this->model->getConditionVariables(), $this->model->getConditionVariableTypes());
+        $totalCount = $this->db->fetchOne(
+            $queryBuilder->getSQL(),
+            $this->model->getConditionVariables() + $queryBuilder->getParameters(),
+            $this->model->getConditionVariableTypes() + $queryBuilder->getParameterTypes()
```

Comment:
Theoretically this is enough

```suggestion
            $queryBuilder->getParameters(),
            $queryBuilder->getParameterTypes()
```

https://github.com/pimcore/pimcore/blob/a0215a0d76ae5e69a61d20b34ca64bc019c2fb4f/lib/Model/Listing/Dao/QueryBuilderHelperTrait.php#L77

`queryBuilder = $this->getQueryBuilder();` -> `applyListingParametersToQueryBuilder` -> `applyConditionsToQueryBuilder` is done before `onCreateQueryBuilderCallback`

The doubt is that it might need to be changed anywhere the `getQueryBuilder` is used, right?